mcs/class/System/System.Net/HttpWebRequest.cs

```diff
@@ -967,29 +948,26 @@ public override Task<Stream> GetRequestStreamAsync ()
 				}
 				throw new WebException (SR.net_timeout, WebExceptionStatus.Timeout);
 			} catch (Exception ex) {
-				throw FlattenException (ex);
+				throw GetWebException (ex, aborted ());
 			} finally {
 				cts.Dispose ();
 			}
 		}
 
 		Task<T> RunWithTimeout<T> (Func<CancellationToken, Task<T>> func)
 		{
-			return RunWithTimeout (func, timeout, Abort);
+			// Call `func` here to propagate any potential exception that it
+			// might throw to our caller rather than returning a faulted task.
+			var cts = new CancellationTokenSource ();
+			var workerTask = func (cts.Token);
+			return RunWithTimeoutWorker (workerTask, timeout, Abort, () => Aborted, cts);
 		}
 
 		async Task<HttpWebResponse> MyGetResponseAsync (CancellationToken cancellationToken)
 		{
 			if (Aborted)
 				throw CreateRequestAbortedException ();
 
-			if (method == null)
```

Comment:
This is impossible, we assign a default value to the field and have null checks in all accessors.